packages/core/crates/core-types/src/acknowledgement.rs

```diff
@@ -182,7 +203,19 @@ impl AcknowledgedTicket {
         let mut computed_ticket_luck = [0u8; 8];
         computed_ticket_luck[1..].copy_from_slice(&self.get_luck(domain_separator).expect("unsigned ticket"));
 
-        u64::from_be_bytes(signed_ticket_luck) <= u64::from_be_bytes(signed_ticket_luck)
+        u64::from_be_bytes(computed_ticket_luck) <= u64::from_be_bytes(signed_ticket_luck)
+    }
+}
+
+impl std::fmt::Display for AcknowledgedTicket {
```

Comment:
This should be removed and replaced with auto-derived `Debug`, but I'm gonna change it in my follow-up PR.